skimage/measure/_regionprops.py

```diff
@@ -517,10 +517,13 @@ def regionprops(label_image, intensity_image=None, cache=True):
 
     """
 
-    label_image = np.squeeze(label_image)
+    squeezed_label_image = np.squeeze(label_image)
 
-    if label_image.ndim not in (2, 3):
-        raise TypeError('Only 2-D and 3-D images supported.')
+    if squeezed_label_image.ndim not in (2, 3):
+        if label_image.ndim not in (2, 3):
+            raise TypeError('Only 2-D and 3-D images supported.')
```

Comment:
I guess the signature should be changed accordingly. At the moment, it is: `label_image : (N, M) ndarray`.